packages/datadog_flutter_plugin/lib/src/rum/rum_gesture_detector.dart

```diff
@@ -0,0 +1,272 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019-Present Datadog, Inc.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+
+import '../../datadog_flutter_plugin.dart';
+
+// The distance a 'pointer' can move and still be considered a tap.
+const _tapSlop = 20;
+const _tapSlopSquared = _tapSlop * _tapSlop;
+
+@immutable
+class _ElementDescription {
+  final Element element;
+  final String elementDescription;
+
+  const _ElementDescription(
+    this.element,
+    this.elementDescription,
+  );
+}
+
+/// Detect simple user actions and send them to RUM.
+///
+/// This wrapper widget automatically detects simple user actions (taps and
+/// swipes) that occur in its tree and sends them to RUM. It detects
+/// interactions with several common Flutter widgets, including
+/// [ElevatedButton], [TextButton], [InkWell], and [GestureDetector].
+///
+/// For most Button types, the detector will look for a [Text] widget child,
+/// which it will use for the description of the action. In other cases, it will
+/// look for a child [Semantics] object, or an [Icon] with its [semanticsLabel]
+/// property set.
+///
+/// Alternately, you can enclose any Widget tree with a
+/// [RumUserActionAnnotation], which will use the provided description when
+/// reporting user actions detected in the child tree, without changing the
+/// Semantics of the tree.
+class RumUserActionDetector extends StatefulWidget {
+  @internal
+  static final elementMap = <RumUserActionDetector, Element>{};
```

Comment:
Yes, it has to be static. Widgets and State are automatically connected by Flutter, and widgets / elements get recreated all the time, whereas State is long lived. A new Widget object will not always create a new State object, only the first time it's constructed.

If the widget tree above the RumUserActionDetector, it will create a new one, which it will connect with the existing StatefulElement / State object and call "didUpdateWidget." But, Widgets and State don't have a good way to get their Elements, so I need a global map of what Widgets map to what Elements, hence the static Map.

Technically, the static map could probably be on the State though.